### PR TITLE
Add a HandleChangeWorld method in the Player handler

### PR DESCRIPTION
**Pull request by Prim69**

### CODE REVIEW

File: server/player/handler.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ type Handler interface {
 	HandleMove(ctx *event.Context, newPos mgl64.Vec3, newYaw, newPitch float64)
 	// HandleTeleport handles the teleportation of a player. ctx.Cancel() may be called to cancel it.
 	HandleTeleport(ctx *event.Context, pos mgl64.Vec3)
+	// HandleChangeWorld handles when the player is added to a new world.
```
Comment (Sandertv) [inline server/player/handler.go:25]: Might be a good idea to document that these worlds may be nil when added to a world for the first time or removed at the end.

**Review by Sandertv — COMMENTED** [review]
One small documentation improvement to be made, this looks good otherwise!

**Comment by Prim69** [comment]
That's done!

File: server/player/handler.go [hunk 2]
```diff
@@ -22,8 +22,8 @@ type Handler interface {
 	HandleMove(ctx *event.Context, newPos mgl64.Vec3, newYaw, newPitch float64)
 	// HandleTeleport handles the teleportation of a player. ctx.Cancel() may be called to cancel it.
 	HandleTeleport(ctx *event.Context, pos mgl64.Vec3)
-	// HandleChangeWorld handles when the player is added to a new world.
-	HandleChangeWorld(old *world.World, new *world.World)
+	// HandleChangeWorld handles when the player is added to a new world. These values may be nil.
```
Comment (T14Raptor) [inline server/player/handler.go:25]: Pretty sure only the new world could be nil here.

Comment (Prim69) [reply server/player/handler.go:25]: What if a world gets deleted?

Comment (Prim69) [reply server/player/handler.go:25]: Unless you mean old, that makes sense

Comment (T14Raptor) [reply server/player/handler.go:25]: No, I mean new, do you mean deleted as in closed?

Comment (T14Raptor) [reply server/player/handler.go:25]: Wait, I mean old, I just confused the shit out of myself

**Comment by Prim69** [comment]
Anything else I should change?

File: server/player/handler.go [hunk 3]
```diff
@@ -22,6 +22,8 @@ type Handler interface {
 	HandleMove(ctx *event.Context, newPos mgl64.Vec3, newYaw, newPitch float64)
 	// HandleTeleport handles the teleportation of a player. ctx.Cancel() may be called to cancel it.
 	HandleTeleport(ctx *event.Context, pos mgl64.Vec3)
+	// HandleChangeWorld handles when the player is added to a new world. oldWorld may be nil.
+	HandleChangeWorld(oldWorld *world.World, newWorld *world.World)
```
Comment (Sandertv) [inline server/player/handler.go:26]: Would go back to calling this just `old` and `new`, the types make it clear these are worlds. Should be good to go after.

Comment (Prim69) [reply server/player/handler.go:26]: new conflicts with Go's `new` function

Comment (Sandertv) [reply server/player/handler.go:26]: Could name them `before` and `after` then.

**Comment by Prim69** [comment]
Could I get another check on this

**Review by Sandertv — APPROVED** [review]
Looks good, thanks for the PR!

**Pull request merged**